File: pset1/greedy.c
/**
 * greedy.c
 * 
 * Natalya Ostiak
 * Technology Nation
 * 
 * Calculates the minimum number of coins required to give a user change.
 */
 
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float n;
    do
    {
        printf("How much change is owed?\n");
        n = GetFloat();
    }
    while(n < 0);
    
    int i = (int) round(n * 100);
    int c = 0;
    
    c = i / 25;
    i = i % 25;
    
    c += i / 10;
    i = i % 10;
     
    c += i / 5;    
    
    printf("%d\n", c += i%5);
}

File: pset6/greedy.py
# greedy.py
# 
# Natalia Ostiak
# Technology Nation
# 
# Calculates the minimum number of coins required to give a user change.

import cs50

while True:
    print("How much change is owed?")
    n = cs50.get_float()
    if n >= 0:
        break
    
i = round(n * 100)

c = i // 25
i = i % 25

c += i//10
i = i % 10
     
c += i//5    

c += i%5
      
print(c)
